src/nvim/os/fsnotification.c

```diff
@@ -0,0 +1,248 @@
+#include <stdbool.h>
+#include <stdlib.h>
+#include <assert.h>
+
+#include <uv.h>
+
+#include "nvim/vim.h"
+#include "nvim/buffer.h"
+#include "nvim/lib/khash.h"
+#include "nvim/lib/klist.h"
+#include "nvim/fileio.h"
+#include "nvim/os/fs_defs.h"
+#include "nvim/os/fsnotification.h"
+#include "nvim/os/event.h"
+
+
+// Apparentely, the destruction mechanism is not very well implemented in klist.
+// Proper resource handling will be done manually
+#define _destroy(x)
+KLIST_INIT(Watcher, Watcher, _destroy)
+#undef _destroy
+
+static klist_t(Watcher)* watchers_list = NULL;
+
+KHASH_MAP_INIT_STR(EventTable, int)
+khash_t(EventTable)* event_lookup = NULL;
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "os/fsnotification.c.generated.h"
+#endif
+
+typedef struct {
+    buf_T* buffer;
+    khiter_t lookup_iter;
+    uv_fs_event_t* handle;
+} EvtData;
```

Comment:
Oh, I see how I was unclear. I was just advising to rename `EvtData` to `EventData`. Also I noticed that the struct should have two space indents.